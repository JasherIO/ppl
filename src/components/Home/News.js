import React from 'react'
import { Link, graphql, StaticQuery } from 'gatsby'
import moment from 'moment'

const Level = () => (
  <div className="level is-mobile">
    <div className="level-left">
      <div className="level-item">
        <p className="title is-4">Recent News</p>
      </div>
    </div>
    <div className="level-right">
      <div className="level-item">
        <Link to="/news" className="button is-primary is-outlined">
          View All News
        </Link>
      </div>
    </div>
  </div>
)

const Card = ({ node: post }) => (
  <div className="column" key={post.id}>
    <div className="card">
      <div className="card-image">
        <figure className="image is-16by9">
          <Link to={post.fields.slug}>
            <img src={post.frontmatter.cover} alt={post.frontmatter.title} />
          </Link>
        </figure>
      </div>
      <div className="card-content">
        <div className="content">
          <div className="title is-4">
            <Link to={post.fields.slug}>
              {post.frontmatter.title}
            </Link>
          </div>
          <div className="subtitle is-6">
            <div className="level is-mobile">
              <div className="level-left">
                <div className="level-item">
                  <span className="has-text-dark is-uppercase has-text-weight-bold">{post.frontmatter.category}</span>
                </div>
              </div>
              <div className="level-right">
                <div className="level-item">
                  <time dateTime={post.frontmatter.date} className="has-text-weight-light">{moment(post.frontmatter.date).fromNow()}</time>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
)

const Section = ({ data }) => {
  const { edges: posts } = data.allMarkdownRemark

  return (
    <section className="section">
      <div className="container">
        <Level />

        <div className="columns">
          {posts.map(Card)}
        </div>
      </div>
    </section>
  )
}

const query = graphql`
  query {
    allMarkdownRemark(
      sort: { order: DESC, fields: [frontmatter___date] },
      filter: { frontmatter: { templateKey: { eq: "blog-post" } }}
      limit: 3
    ) {
      edges {
        node {
          id
          fields {
            slug
          }
          frontmatter {
            title
            cover
            category
            date(formatString: "MMMM DD, YYYY")
          }
        }
      }
    }
  }
`

export default props => (
  <StaticQuery query={query} render={data => <Section data={data} {...props} />} />
)
