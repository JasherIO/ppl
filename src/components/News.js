import React from 'react'
import { Link, graphql, StaticQuery } from 'gatsby'
import Cards from './News-Cards'

export const Title = () => (
  <div className="level is-mobile">
    <div className="level-left">
      <div className="level-item">
        <p className="title is-4">Recent News</p>
      </div>
    </div>
    <div className="level-right">
      <div className="level-item">
        <Link to="/news" className="button is-primary is-outlined">
          View All News
        </Link>
      </div>
    </div>
  </div>
)

export const PureSection = ({ data }) => {
  const { edges: posts } = data.allMarkdownRemark

  return (
    <section className="section background is-hexellence" style={{ paddingBottom: "5rem" }}>
      <div className="container">
        <Title />

        <Cards posts={posts} />
      </div>
    </section>
  )
}

const query = graphql`
  query {
    allMarkdownRemark(
      sort: { order: DESC, fields: [frontmatter___date] },
      filter: { frontmatter: { templateKey: { eq: "post" } }}
      limit: 3
    ) {
      edges {
        node {
          id
          fields {
            slug
          }
          frontmatter {
            title
            cover {
              childImageSharp {
                fluid(maxWidth: 700) {
                  ...GatsbyImageSharpFluid_withWebp
                }
              }
            }
            category
            date
          }
        }
      }
    }
  }
`

export const Section = props => (
  <StaticQuery query={query} render={data => <PureSection data={data} {...props} />} />
)

export default Section
