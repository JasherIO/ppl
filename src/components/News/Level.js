import React from 'react'

const Level = () => (
  <div className="level is-mobile">
    <div className="level-left">
      <div className="level-item">
        <h1 className="title is-3">Recent News</h1>
      </div>
    </div>
    {/* <div className="level-right">
      <div className="level-item">
        <div className="field is-horizontal">
          <div className="field-label is-normal">
            <label className="label">Category</label>
          </div>
          <div className="field-body">
            <div className="field">
              <div className="select is-primary">
                <select>
                  <option>News</option>
                  <option>Review</option>
                  <option>Video</option>
                </select>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div> */}
  </div>
)

export default Level