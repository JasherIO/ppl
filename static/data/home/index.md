---
templateKey: home-post
tabs:
  - name: Players
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus facilisis, nisl quis porta bibendum, nulla purus vestibulum sem, sit amet ultrices nisi elit ut odio. Sed rutrum eu velit at rhoncus.'
  - name: Teams
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus facilisis, nisl quis porta bibendum, nulla purus vestibulum sem, sit amet ultrices nisi elit ut odio. Sed rutrum eu velit at rhoncus.'
  - name: Format
    description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus facilisis, nisl quis porta bibendum, nulla purus vestibulum sem, sit amet ultrices nisi elit ut odio. Sed rutrum eu velit at rhoncus.'
sponsors:
  - img: /img/AllYours.png
    link: 'https://google.com'
    name: All Yours Buffet
  - img: /img/WhatASave.png
    link: 'https://google.com'
    name: What a Save Esports
  - img: /img/EggingGrocers.png
    link: 'https://google.com'
    name: Egging Grocers
---

