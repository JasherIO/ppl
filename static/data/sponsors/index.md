---
templateKey: sponsor-post
sponsors:
  - img: /img/AllYours.png
    link: 'https://google.com'
    name: All Yours Buffet
  - img: /img/WhatASave.png
    link: 'https://google.com'
    name: What a Save Esports
  - img: /img/EggingGrocers.png
    link: 'https://google.com'
    name: Egging Grocers
---

