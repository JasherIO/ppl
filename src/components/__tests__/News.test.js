import React from "react"
import renderer from "react-test-renderer"

import { PureSection, Title } from "../News"
import data from './posts.json'

describe("Title", () =>
  it("renders correctly", () => {
    const tree = renderer.create(<Title />).toJSON()
    expect(tree).toMatchSnapshot()
  }))

jest.mock('../../components/News-Cards', () => () => (<></>))

describe("News Section", () =>
  it("renders correctly", () => {
    const tree = renderer
      .create(<PureSection data={data} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  }))