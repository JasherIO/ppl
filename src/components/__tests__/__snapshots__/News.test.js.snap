// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Section renders correctly 1`] = `
<section
  className="section background is-hexellence"
  style={
    Object {
      "paddingBottom": "5rem",
    }
  }
>
  <div
    className="container"
  >
    <div
      className="level is-mobile"
    >
      <div
        className="level-left"
      >
        <div
          className="level-item"
        >
          <p
            className="title is-4"
          >
            Recent News
          </p>
        </div>
      </div>
      <div
        className="level-right"
      >
        <div
          className="level-item"
        >
          <a
            className="button is-primary is-outlined"
            href="/news"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            View All News
          </a>
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Title renders correctly 1`] = `
<div
  className="level is-mobile"
>
  <div
    className="level-left"
  >
    <div
      className="level-item"
    >
      <p
        className="title is-4"
      >
        Recent News
      </p>
    </div>
  </div>
  <div
    className="level-right"
  >
    <div
      className="level-item"
    >
      <a
        className="button is-primary is-outlined"
        href="/news"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        View All News
      </a>
    </div>
  </div>
</div>
`;
